Tests/OpenAPIRuntimeTests/Interface/Test_HTTPBody.swift

```diff
@@ -0,0 +1,189 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftOpenAPIGenerator open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftOpenAPIGenerator project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftOpenAPIGenerator project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+import XCTest
+@_spi(Generated)@testable import OpenAPIRuntime
+import Foundation
+
+final class Test_Body: Test_Runtime {
```

Comment:
There are no error tests for iterating a sequence more than once?

Reply:
Whether `HTTPBody` as an async sequence can be iterated multiple times depends on the creator of it. Either they provide a type that is guaranteed to be iterable multiple times (e.g. a sync collection), or they provide `iterationBehavior` explicitly.

If the consumer ignores the declared iteration behavior and iterates the body multiple times, even if it declared it can only be iterated once, we will [crash](https://github.com/apple/swift-openapi-runtime/pull/47/files#diff-e089a09ba06f85ed1377a6d7ec2a0c36a9f91a3d1b276134aa3156ec6484cedeR384) with a descriptive error. Since `makeAsyncIterator` isn't a throwing function, we can't throw an error, so crashing is the only choice here if we want to provide a good error. However, if someone tries to use the `collecting` initializers that we provide on types like `String`, those at least [throw an error](https://github.com/apple/swift-openapi-runtime/pull/47/files#diff-e089a09ba06f85ed1377a6d7ec2a0c36a9f91a3d1b276134aa3156ec6484cedeR443) when attempting to iterate the body multiple times, instead of crashing.

Reply:
Agreed, I surfaced a property just for testing allowing for better tests: https://github.com/apple/swift-openapi-runtime/pull/47/commits/72d241d0f22a9ac11482213b3febc0f8d24837b9